lms/djangoapps/discussion/tasks.py

```diff
@@ -90,3 +93,26 @@ def _get_thread_url(context):
         'id': context['thread_id'],
     }
     return urljoin(settings.LMS_ROOT_URL, permalink(thread_content))
```

Comment:
A couple of notes:

1. I think you need to customize LMS_ROOT_URL to be white label site aware.
2. You will need to escape any course IDs in this URL if you are planning on using Sailthru's GA link re-writing. If not, I recommend appending UTM parameters to this URL so that you can track clicks.

Reply:
We're not planning to use Sailthru's GA link rewriting, but looking at the strings that this method produces, I think it's already escaping the course key.